### PR TITLE
cleanup manual value clamping: sdl2_to_1_2_backports.cpp

### PR DESCRIPTION
**Pull request by bcko**
part of #1666

### CODE REVIEW

File: Source/utils/sdl2_to_1_2_backports.cpp [hunk 1]
```diff
@@ -449,24 +450,16 @@ int SDL_BlitScaled(SDL_Surface *src, SDL_Rect *srcrect,
 
 	final_src.x = (Sint16)SDL_floor(src_x0 + 0.5);
 	final_src.y = (Sint16)SDL_floor(src_y0 + 0.5);
-	src_w = (int)SDL_floor(src_x1 + 1 + 0.5) - (int)SDL_floor(src_x0 + 0.5);
-	src_h = (int)SDL_floor(src_y1 + 1 + 0.5) - (int)SDL_floor(src_y0 + 0.5);
-	if (src_w < 0)
-		src_w = 0;
-	if (src_h < 0)
-		src_h = 0;
+	src_w = std::max((int)SDL_floor(src_x1 + 1 + 0.5) - (int)SDL_floor(src_x0 + 0.5), 0);
```
Comment (ephphatha) [inline Source/utils/sdl2_to_1_2_backports.cpp:453]: These calls to sdl_floor should be cleaned up. The only time a non-integer value is relevant is when determining and using the scaling_w/h variables, so with a bit of work these casts and calls to SDL_floor could be removed.

If you're willing to go through this code it'd be great to clean up the whole function. For now just changing the cast on these lines to c++ style static_casts would be ideal.

Comment (bcko) [reply Source/utils/sdl2_to_1_2_backports.cpp:453]: Updated the pull request to change (int) to static_cast<int>. Thank you @ephphatha !

At the moment, I might not know enough of c++, SDL, and devilutionx codebase to properly clean up the whole function. As I gain more knowledge, I am happy to revisit so I can clean up the whole function. :)

Comment (ephphatha) [reply Source/utils/sdl2_to_1_2_backports.cpp:453]: you got the bottom two casts but the four on this line and the next can use static_cast as well :)

**Review by ephphatha — CHANGES_REQUESTED** [review]

File: Source/utils/sdl2_to_1_2_backports.cpp (hunk 1 above)
Comment (bcko) [reply Source/utils/sdl2_to_1_2_backports.cpp]: Updated the pull request. I am learning a lot from your pull request reviews. Thank you @ephphatha ! :).

**Review by ephphatha — APPROVED** [review]

**Pull request merged**